src/test/java/duke/logic/ParserTest.java

```diff
@@ -0,0 +1,145 @@
+package duke.logic;
+
+import duke.command.*;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.time.LocalDate;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+public class ParserTest {
+    private static LocalDate date = LocalDate.now();
+    private static TaskList taskList = new TaskList(new Storage("./test/parsertest.txt"));
+    static {
+        taskList.add(new ToDo("todo"));
+        taskList.add(new Deadline("deadline", date));
+        taskList.add(new Event("event", date));
+    }
+
+    @Test
+    public void parse() {
+        Parser parser = new Parser(ParserTest.taskList);
+
+        //test bye
+        assertTrue(parser.parse("bye") instanceof ByeCommand);
+
+        //test mark
+        assertTrue(parser.parse("mark 2") instanceof MarkCommand);
+
+        //test unmark
+        assertTrue(parser.parse("unmark 2") instanceof UnmarkCommand);
+
+        //test delete
+        assertTrue(parser.parse("delete 2") instanceof DeleteCommand);
+
+        //test todo
+        assertTrue(parser.parse("todo desc") instanceof ToDoCommand);
+
+        //test deadline
+        assertTrue(parser.parse("deadline desc /by 1/1/22") instanceof DeadlineCommand);
+
+        //test event
+        assertTrue(parser.parse("event desc /at 1/1/22") instanceof EventCommand);
+
+    }
+
+    @Test
+    public void parse_badInput_printMessage() {
```

Comment:
Good test case naming! This will make readers understand what the testcase does very quickly 👍🏻